### PR TITLE
releaser: Add Vitest coverage for utility functions

### LINKED ISSUE #798: Add Vitest coverage for tools/releaser utility functions
**Summary**

The tools/releaser package is responsible for several critical release-management tasks within the monorepo, including version handling, git tag management, changelog generation, and ArtifactHub synchronization.

While exploring the codebase, I noticed that the utility functions under tools/releaser/src/utils currently have little to no unit test coverage. Since these utilities are part of the release pipeline, regressions in their behavior could potentially lead to incorrect versioning or release artifacts.

**Proposal**

Add Vitest-based unit tests for the core utility functions in tools/releaser/src/utils.

Potential areas to cover include:

1) Version sanitization and validation logic
2) Git log parsing and commit extraction
3) Version comparison utilities
4) Release metadata generation
5) Edge cases and malformed input handling

I'd be happy to work on an initial PR for the utility test coverage if this proposal sounds reasonable.

### PR DESCRIPTION
**Pull request by skools-here**
## Summary

Adds Vitest test coverage for the releaser utility modules.

fixes : #798
### Changes

- Add tests for ArtifactHub utilities (`artifacthub.ts`)
- Add tests for Git utilities (`git.ts`)
- Add tests for GitHub utilities (`github.ts`)
- Add tests for plugin utilities (`plugin.ts`)
- Add tests for version utilities (`version.ts`)
- Add Vitest coverage configuration

### Results

- 62 tests added
- All tests passing

### Note for reviewer
- This PR introduces a Node engines constraint (`^20.19.0 || >=22.13.0`) in `tools/releaser/package.json`. Please make sure your local development environment and CI runners are using a supported Node.js version.
- Upgrade `inquirer` from `^8.2.6` to `^14.0.2`
- Remove `@types/inquirer` since `inquirer` v14 ships with built-in TypeScript types.

### CODE REVIEW

File: tools/releaser/src/utils/version.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { sanitizeVersion, validateVersion } from './version.js';
```
[the remark this answers is not in the document]
Comment (skools-here) [reply tools/releaser/src/utils/version.test.ts]: The package uses Node16 ESM ("type": "module" and "moduleResolution": "node16"). The .js import extensions are intentional and required for correct TypeScript/Node ESM resolution, so no changes are required here according to me.

**Comment by skools-here** [comment]
Hi! @illume  I've fixed the Copilot review comments . Could you please take another look when you have a chance?

Thanks!

**Review by illume — CHANGES_REQUESTED** [review]
Thanks for the contribution.


Could you take a look at the commit messages in this PR? We follow a Linux kernel style for git commits — see the [contributing guide](https://headlamp.dev/docs/latest/contributing#2-follow-commit-guidelines) and `git log` for examples.

<details>
<summary>Commits that need attention</summary>

- `releaser: address ESM import review comments` — Description must start with a capital letter — e.g. `frontend: HomeButton: Fix the button` not `frontend: HomeButton: fix the button`.
- `releaser: align tests with ESM import conventions` — Description must start with a capital letter — e.g. `frontend: HomeButton: Fix the button` not `frontend: HomeButton: fix the button`.
- `releaser: improve test typing and mocking` — Description must start with a capital letter — e.g. `frontend: HomeButton: Fix the button` not `frontend: HomeButton: fix the button`.

</details>

<details>
<summary>Commit guidelines</summary>

- Use atomic commits focused on a single change.
- Use the title format `<area>: <Description of changes>` — description must start with a capital letter.
- Keep the title under 72 characters (soft requirement).
- Explain the intention and why the change is needed.
- Make commit titles meaningful and describe what changed.
- Do not add code that a later commit rewrites; squash or reorder commits instead.
- Do not include `Fixes #NN` in commit messages.

Good examples:

- `frontend: HomeButton: Fix so it navigates to home`
- `backend: config: Add enable-dynamic-clusters flag`

</details>


There are some open Copilot review comments — could you take a look at them? Please mark each one as resolved once you've addressed it.

**Comment by skools-here** [comment]
ptal @illume

**Review by illume — CHANGES_REQUESTED** [review]
For the failing security CI check... the dependencies mentioned are:`Introduced through inquirer@8.2.6 Fixed in tmp@0.2.4`

**Comment by skools-here** [comment]
ptal @illume

**Comment by skools-here** [comment]
ptal @illume @itvi-1234
